src/Engines/engine.ts

```diff
@@ -391,9 +391,9 @@ export class Engine extends ThinEngine {
     public isPointerLock = false;
 
     /**
-     * Stores instance of DeviceInputSystem
+     * @hidden
      */
-    public deviceInputSystem: DeviceInputSystem;
+    public _deviceSourceManager: InternalDeviceSourceManager;

```

Comment:
This seems backwards to me. Why not augment the engine from the DeviceSourceManager side and then you won't have to expose InternalDeviceSourceManager at all. You can use engine.onDisposeObservable to dispose the instance.

Reply:
Discussed offline